apps/ex_wire/lib/ex_wire/struct/routing_table.ex

```diff
@@ -0,0 +1,151 @@
+defmodule ExWire.Struct.RoutingTable do
+  @moduledoc """
+  Module for working with current node's buckets
+  """
+
+  alias ExWire.Struct.{Bucket, Peer}
+  alias ExWire.KademliaConfig
+
+  @network Application.fetch_env(:ex_wire, :network_process_name)
+
+  defstruct [:current_node, :buckets]
+
+  @type t :: %__MODULE__{
+          current_node: Peer.t(),
+          buckets: [Bucket.t()]
+        }
+
+  @doc """
+  Creates new routing table.
+
+  ## Examples
+
+      iex> node = ExWire.Struct.Peer.new("13.84.180.240", 30303, "6ce05930c72abc632c58e2e4324f7c7ea478cec0ed4fa2528982cf34483094e9cbc9216e7aa349691242576d552a2a56aaeae426c5303ded677ce455ba1acd9d", time: :test)
+      iex> table = node |> ExWire.Struct.RoutingTable.new()
+      iex> table.buckets |> Enum.count
+      256
+  """
+  @spec new(Peer.t()) :: t()
+  def new(peer = %Peer{}) do
+    initial_buckets = initialize_buckets()
+
+    %__MODULE__{
+      current_node: peer,
+      buckets: initial_buckets
+    }
+  end
+
+  @doc """
+  Returns table's buckets.
+  """
+  @spec buckets(t()) :: [Bucket.t()]
+  def buckets(%__MODULE__{buckets: buckets}), do: buckets
+
+  @doc """
+  Adds node to routing table.
+  """
+  @spec add_node(t(), Peer.t()) :: t()
+  def add_node(
+        table = %__MODULE__{current_node: %Peer{remote_id: current_node_id}},
+        %Peer{remote_id: current_node_id}
+      ),
+      do: table
+
+  def add_node(table = %__MODULE__{}, node = %Peer{}) do
```

Comment:
tbh I'm not sure if its actually a good idea to call this method `add_node`.
As we know it doesn't always "add" node, according to the "Kademlia" paper it
> updates the appropriate _k_-bucket for the senders node ID

 so maybe it would be better to call it `update_sender`.
What do you think? Same thing regarding the `add_node` function in `Bucket`:
maybe `update_node` better reveals the intention there too.

Reply:
I think `add_node` is ok. Because users of this method do not if a node is already in the routing table or not.

Reply:
yeah, I understand that, but I still don’t see how it could be a reason to name it add_node while its purpose is to refresh the node in k bucket :) maybe I didn’t understand the point